apps/web/src/pages/Pools/EditPoolPage/components/AssetSkillsSection.tsx

```diff
@@ -3,40 +3,91 @@ import { useIntl } from "react-intl";
 import AcademicCapIcon from "@heroicons/react/24/outline/AcademicCapIcon";
 
 import { ToggleSection } from "@gc-digital-talent/ui";
-import { PoolStatus, Skill, UpdatePoolInput } from "@gc-digital-talent/graphql";
+import { notEmpty } from "@gc-digital-talent/helpers";
+import {
+  PoolSkillType,
+  SkillCategory,
+  SkillLevel,
+  Pool,
+  PoolStatus,
+  Skill,
+} from "@gc-digital-talent/graphql";
 
 import useToggleSectionInfo from "~/hooks/useToggleSectionInfo";
+import { EditPoolSectionMetadata } from "~/types/pool";
 
 import SkillTable from "./SkillTable";
-import { SectionProps } from "../types";
+import { PoolSkillMutationsType } from "../types";
 
-export type AssetSkillsSubmitData = Pick<UpdatePoolInput, "nonessentialSkills">;
-type AssetSkillsSectionProps = SectionProps<AssetSkillsSubmitData> & {
+type AssetSkillsSectionProps = {
+  pool: Pool;
+  sectionMetadata: EditPoolSectionMetadata;
   skills: Array<Skill>;
+  poolSkillMutations: PoolSkillMutationsType;
 };
 
 const AssetSkillsSection = ({
   pool,
   skills,
   sectionMetadata,
-  onSave,
+  poolSkillMutations,
 }: AssetSkillsSectionProps): JSX.Element => {
   const intl = useIntl();
   const { icon } = useToggleSectionInfo({
     isNull: false,
     emptyRequired: false,
     fallbackIcon: AcademicCapIcon,
   });
-  const defaultSkills = pool.nonessentialSkills ? pool.nonessentialSkills : [];
 
-  const handleSave = async (ids: string[]) => {
-    onSave({
-      nonessentialSkills: {
-        sync: ids,
-      },
+  const poolSkills = pool.poolSkills ?? [];
+  const nonessentialPoolSkills = poolSkills
+    .filter(notEmpty)
+    .filter(
+      (poolSkill) =>
+        poolSkill.type === PoolSkillType.Nonessential &&
+        poolSkill.skill &&
+        poolSkill.skill.category &&
+        poolSkill.skill.id,
```

Comment:
Is just checking the type not sufficient? 🤔

Reply:
Leftover frustration from Typescript type checks (Typescript I did filter out undefined skills, skill literally can't be falsy)
Cleared out in https://github.com/GCTC-NTGC/gc-digital-talent/pull/9503/commits/aebd7ff8630d6833d25b09c88737591023ef779d